### PR TITLE
Porting sonic-telemetry dail-in and dial-out test case to be perform by sonic-mgmt

### PR DESCRIPTION
**Pull request by MuLinForest**
### Description of PR

The tests in sonic-telemetry are tested in local machine, therefore I modify the tests to verify the telemetry functionality of real switch. (insecure mode)
Installing go, libyang, libyang-dev in ptf container is required for this testcase.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?
1. Add arguments for telemetry go test to specify the target IP for dail-in mode and the dail-out client IP for dail-out mode.
2. Install go and libyang related debs in ptf container
3. Add a new ansible play for telemetry to prepare test requirement, transform the test data, and perform the go test.
#### How did you verify/test it?
Test the response from sonic-telemetry service
#### Any platform specific information?
N/A
#### Supported testbed topology if it's a new test case?
ptf32, ptf64

### CODE REVIEW

**Comment by lguohan** [comment]
thanks for you contribution. but we no longer accepting new testcases written in ansible-yml. can you rewrite the test case in pytest-ansible. example can be found in tests/ folder.

File: ansible/roles/test/tasks/telemetry.yml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    get_url:
+      url: https://dl.google.com/go/{{ go_fileName }}
+      dest: "{{ path_tmp_telemetry }}/{{ go_fileName }}"
+  - name: Unarchive Go
+    delegate_to: localhost
+    become: true
+    unarchive:
+      src: "{{ path_tmp_telemetry }}/{{ go_fileName }}"
+      dest: /usr/local/
+      remote_src: yes
```
Comment (lguohan) [inline ansible/roles/test/tasks/telemetry.yml]: the go binary should be built into the docker-ptf directly in the sonic-buildimage repo.

File: ansible/roles/test/tasks/telemetry.yml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    delegate_to: localhost
+    become: true
+    apt:
+      name: python3-pip
+  - name: pip jinja2
+    delegate_to: localhost
+    become: true
+    pip:
+      name: jinja2>=2.10
+      executable: pip3
```
Comment (lguohan) [inline ansible/roles/test/tasks/telemetry.yml]: i do not know why this is needed? i think they are in the sonic-mgmt docker.

File: ansible/roles/test/tasks/telemetry.yml (hunk 1 above)
Comment (MuLinForest) [reply ansible/roles/test/tasks/telemetry.yml]: The go binary here is in sonic-mgmt container to make (Makefile in sonic-telemetry) the dependency for running telemetry test

File: ansible/roles/test/tasks/telemetry.yml (hunk 2 above)
Comment (MuLinForest) [reply ansible/roles/test/tasks/telemetry.yml]: sonic-mgmt-framework/tools/ui_gen/src/uigen.py needs jinja2 from python3

**Comment by lguohan** [comment]
the pr is a little bit large especially for the go program part. will you be ok to discuss this test design in our next sonic test subgroup meeting?

**Comment by gord1306** [comment]
The idea we provide just to move the original go test cases which in the sonic-telemetry [(link)](https://github.com/Azure/sonic-telemetry/blob/master/doc/grpc_telemetry.md) can be performed in the sonic-mgmt. Therefore the gNMI utilities are required to be put in to the image

**Pull request closed without merging**